src/Models/LagrangianParticleTracking/lagrangian_particle_advection.jl

```diff
@@ -6,8 +6,17 @@ using Oceananigans.Models: total_velocities
 #####
 
 # Functions for bouncing particles off walls to the right and left
-@inline  bounce_left(x, xᴿ, Cʳ) = xᴿ - Cʳ * (x - xᴿ)
-@inline bounce_right(x, xᴸ, Cʳ) = xᴸ + Cʳ * (xᴸ - x)
+@inline function bounce_left(x, xᴸ, xᴿ, Cʳ)
+    xᵢ = xᴿ - Cʳ * (x - xᴿ)
+    # Limit the particle to the left boundary when it leaves from there
+    return ifelse(xᵢ > xᴸ, xᵢ, xᴸ)
```

Comment:
```suggestion
    return ifelse(xᵢ < xᴸ, xᴸ, xᵢ)
```

I think this logic is a little more clear, as in

"if [bad things happen] then [do something special] else [proceed as normal]"